### PR TITLE
should not require json serializer unless it is configured.

### PR DESCRIPTION
**Pull request by fugufish**
this isn't really testable, its more of a require/loading detail

### CODE REVIEW

File: lib/polyn/application.rb [hunk 1]
```diff
@@ -72,10 +74,10 @@ def initialize(options)
...
     def publish(type, payload)
       event = Event.new({
-        type:   type,
-        source: "#{source_prefix}.#{name}",
-        data:   payload,
-      })
+                          type:   type,
+                          source: "#{source_prefix}.#{name}",
+                          data:   payload,
+                        })
```
Comment (reecebisel) [inline lib/polyn/application.rb:81]: I do not like this styling change. It doesn't change anything. But I am not a fan.

File: lib/polyn/application.rb [hunk 2]
```diff
@@ -86,8 +88,18 @@ def on_message((msg, *args))
...
+                         " Check logs for details.")
+          transit.shutdown
+          reactor_manager&.shutdown
+        ensure
+          puts "Polyn forced the application to exit due to an error."
+          Process.kill("INT", Process.pid)
+        end
       else
-        raise ArgumentError, "unknown message: #{msg.inspect}"
+        pass
```
Comment (reecebisel) [inline lib/polyn/application.rb:102]: Why was this changed to pass?

File: lib/polyn/application.rb [hunk 3]
```diff
@@ -86,8 +88,18 @@ def on_message((msg, *args))
...
+      when :terminated
+        begin
+          logger.fatal("there was a problem with initializing the Polyn application, shutting down" \
+                         " Check logs for details.")
+          transit.shutdown
+          reactor_manager&.shutdown
+        ensure
+          puts "Polyn forced the application to exit due to an error."
+          Process.kill("INT", Process.pid)
+        end
```
Comment (reecebisel) [inline lib/polyn/application.rb:100]: This should be moved to a private method. Its an implementation detail. I am not a huge fan of case statements for this reason.

File: lib/polyn/reactor_manager.rb [hunk 4]
```diff
@@ -20,20 +20,48 @@
...
+      def shutdown
+        actor.ask!(:shutdown)
+      rescue Concurrent::Actor::ActorTerminated
+        logger.warn("reactor manager was already terminated")
+      end
+
+      private
+
+      attr_reader :actor
+    end
```
Comment (reecebisel) [inline lib/polyn/reactor_manager.rb:50]: Any reason for keeping this inside the reactor manager file?

File: lib/polyn/reactor_manager.rb [hunk 5]
```diff
@@ -20,20 +20,48 @@
...
+
     include SemanticLogger::Loggable
 
     ##
     # @private
     def self.spawn(services)
-      super(:service_manager, services)
+      Wrapper.new(super(:reactor_manager, services))
```
Comment (reecebisel) [inline lib/polyn/reactor_manager.rb:57]: Would we need to do some sort of memorization around this?

File: lib/polyn/serializers.rb [hunk 6]
```diff
@@ -39,6 +38,7 @@ def self.for(config)
 
       case type
       when :json
+        require "polyn/serializers/json"
```
Comment (reecebisel) [inline lib/polyn/serializers.rb:41]: Why though?

File: lib/polyn/transit.rb [hunk 7]
```diff
@@ -48,59 +56,69 @@ def self.spawn(*args)
...
         publish(*args)
       when :receive
         receive(*args)
+      when :shutdown
+        logger.warn("shutting down")
+        begin
+          transporter.disconnect!
+        rescue Concurrent::Actor::ActorTerminated
+          logger.warn("transporter already terminated")
+        end
```
Comment (reecebisel) [inline lib/polyn/transit.rb:91]: ditts on the private method.

File: lib/polyn/transit.rb [hunk 8]
```diff
@@ -48,59 +56,69 @@ def self.spawn(*args)
...
     end
 
     private
 
-    attr_reader :transporter, :serializer, :origin, :service_manager
+    attr_reader :transporter, :serializer, :origin, :reactor_manager
 
     def events
-      @events ||= service_manager.ask!(:services).map { |s| s.events.values }.flatten
+      @events ||= reactor_manager.reactors.map { |s| s.events.values }.flatten
```
Comment (reecebisel) [inline lib/polyn/transit.rb:102]: Noice!

File: lib/polyn/transit.rb [hunk 9]
```diff
@@ -48,59 +56,69 @@ def self.spawn(*args)
...
     def pool
       @pool ||= Concurrent::ThreadPoolExecutor.new({
-        min_threads: 5,
-        max_threads: 10,
-        max_queue:   20,
-      })
+                                                     min_threads: 5,
+                                                     max_threads: 10,
+                                                     max_queue:   20,
+                                                   })
```
Comment (reecebisel) [inline lib/polyn/transit.rb:121]: Ditts on the style change here.

File: lib/polyn/transporters/internal.rb [hunk 10]
```diff
@@ -32,6 +32,9 @@ def acknowledge; end
 
       def initialize(transit, options = {})
         super(transit, options)
+        if Object.const_defined?("::Rails")
+          logger.warn("WARNING, the internal transporter is not supported within Rails.")
+        end
```
Comment (reecebisel) [inline lib/polyn/transporters/internal.rb:37]: If we are calling this `polyn-ruby` we shouldn't be checking for rails. Otherwirse we should call this gem `polyn-rails`

File: lib/polyn/utils/generic_concurrent_logger.rb [hunk 11]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
```
Comment (reecebisel) [inline lib/polyn/utils/generic_concurrent_logger.rb:1]: Do we have tests for this?

**Review by reecebisel — CHANGES_REQUESTED** [review]

**Comment by brandynbennett** [comment]
A lot has changes since this, so probably easier to close

**Pull request closed without merging**